erts/emulator/beam/jit/x86/instr_guard_bifs.cpp

```diff
@@ -162,6 +162,72 @@ void BeamModuleAssembler::emit_bif_element(const ArgVal &Fail,
                      signed_val(Pos.getValue()) > 0 &&
                      signed_val(Pos.getValue()) <= (Sint)MAX_ARITYVAL;
 
+    /*
+     * Try to optimize the use of a tuple as a lookup table.
+     */
+    if (exact_type(Pos, BEAM_TYPE_INTEGER) && Tuple.isLiteral()) {
+        Eterm tuple = beamfile_get_literal(beam, Tuple.getValue());
+        if (is_tuple(tuple)) {
+            Label error = a.newLabel(), next = a.newLabel();
+            Sint size = Sint(arityval(*tuple_val(tuple)));
+            auto [min, max] = getIntRange(Pos);
+            bool is_bounded = min <= max;
+            bool can_fail = is_bounded || min < 1 || size < max;
+
+            comment("skipped tuple test since source is always a literal "
+                    "tuple");
+            mov_arg(ARG2, Tuple);
+            mov_arg(ARG1, Pos);
+            x86::Gp boxed_ptr = emit_ptr_val(ARG3, ARG2);
+            a.lea(ARG4, emit_boxed_val(boxed_ptr));
```

Comment:
Can't we just load the untagged tuple pointer directly into ARG4?

Reply:
There does not seem to be an easy way to do that.

The literal is not at its final position, so the instruction that loads the tuple pointer into `ARG2` will be patched later. To load an untagged pointer, we would have to extend the patch machinery in the loader to support untagging a pointer while patching.

Reply:
Ah, yes of course. I had forgotten about that.